tools/clang-format.sh

```diff
@@ -77,10 +77,16 @@ EOF
       echo "or alternatively, undefine the FORMAT environment variable"
       exit 1
   else
+      tmpfile=/tmp/clang-format$$
       for file in $(find $DIR -iname \*.[ch] -o -iname \*.cc -o -iname \*.h.in); do
-    echo $file
-    ${FORMAT} -i $file
+          echo $file
+          ${FORMAT} $file >| $tmpfile
+          if ! cmp --silent $file $tmpfile
+          then
+              \mv $tmpfile $file
+          fi
       done
+      \rm -f $tmpfile
```

Comment:
An alternative would be:
```
      for file in $(find $DIR -iname \*.[ch] -o -iname \*.cc -o -iname \*.h.in); do
          echo $file
          tmpfile=${file}.clang-format$$
          ${FORMAT} $file >| $tmpfile
          if ! cmp --silent $file $tmpfile
          then
              \mv $tmpfile $file
          else
              \rm -f $tmpfile
          fi
      done
```
The advantage is that the moves would be renames, so faster.  The disadvantage is, if you control-C while doing this, the potential leftover garbage tempfile is in the source directory (where it has to get cleaned up manually) rather than under /tmp .

Reply:
That's trivial to deal with, just catch the "kill" signal and cleanup the tmp file.